web/html/src/manager/content-management/project/project.js

```diff
@@ -73,7 +73,11 @@ const Project = (props: Props) => {
   }
 
   const isProjectEdited = changesToBuild.length > 0;
-  const isBuildDisabled = !hasEditingPermissions || _isEmpty(project.environments) || _isEmpty(project.softwareSources);
+  const isBuildDisabled = !hasEditingPermissions
+        || _isEmpty(project.environments)
+        || _isEmpty(project.softwareSources)
+        || project.environments[0].status === "building"
+        || (project.environments[1] || {}).status === "building";
```

Comment:
IIUC those two conditions stand for:
1. if the first environment is building from sources, I will not be able to `Build` because if I do that it would overwrite the current building
2. if the second environment is building, I will not be able to `Build` because if I do that the current building from the  first environment to the second will be overwritten


Are these correct? I'd say let's add one line comment for each, what do you think?

Reply:
Just to elaborate on the 2nd point:
If the 2nd environment is building (in other words promotion from 1st to 2nd env is going on), channels are cloned from the 1st environment to the 2nd one. We want to prevent build, as it would change channels in the 1st environment, which are used by the ongoing promotion.

An example:
```
Initial configuration (source + 2 environments):
S -> 1 -> 2

Lets promote 1 to 2
S -> [ 1 -> 2 ]

Lets  build the source at the same time! 
Wait, we can't, since it's target (1) is already used as a source for promoting.
[S -> [ 1 ]-> 2 ]
```

I'll add some comments, that'd make sense.